templates/nsq/0/docker-compose.yml.tpl

```diff
@@ -0,0 +1,43 @@
+nsqlookupd01:
+  image: nsqio/nsq
+  command: /nsqlookupd
+nsqlookupd02:
+  image: nsqio/nsq
+  command: /nsqlookupd
+nsqlookupd03:
+  image: nsqio/nsq
+  command: /nsqlookupd
+nsqd:
+  image: nsqio/nsq
+  command:
+  - /bin/sh
+  - -c
+  - nsqd --data-path=/data --lookupd-tcp-address=nsqlookupd01:4160 --lookupd-tcp-address=nsqlookupd02:4160 --lookupd-tcp-address=nsqlookupd03:4160 -broadcast-address=$$HOSTNAME
+  labels:
+      io.rancher.scheduler.affinity:host_label_soft: nsqd=true
+      io.rancher.scheduler.affinity:container_label_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
+      io.rancher.sidekicks: data
+      io.rancher.container.hostname_override: container_name
+  volumes_from:
+  - data
+nsqadmin:
+  image: nsqio/nsq
+  command: /nsqadmin --lookupd-http-address=nsqlookupd01:4161 --lookupd-http-address=nsqlookupd02:4161 --lookupd-http-address=nsqlookupd03:4161
+nsq-lb:
+  image: rancher/lb-service-haproxy:v0.7.9
+  ports:
+  - 4150:4150/tcp
+  - 4151:4151/tcp
+  - 4171:4171/tcp
```

Comment:
What do you think to set variables for publish ports??
```
-  - 4150:4150/tcp
-  - 4151:4151/tcp
-  - 4171:4171/tcp
+  - ${NSQD_TCP}:4150/tcp
+  - ${NSQD_HTTP}:4151/tcp
+  - ${NSQADMIN_HTTP}:4171/tcp
```

Reply:
the nsqd servers register themself at the nsqlookupd daemons with their broadcast name and port, i think when we use a different port external it could be a problem. I recommend to use default ports here.

Reply:
Just a proposal, if it's not posiblem no problem. :)